packages/flutter_tools/lib/src/commands/attach.dart

```diff
@@ -359,6 +359,7 @@ known, it can be explicitly provided to attach via the command-line, e.g.
           )
             ..setupTerminal()
             ..registerSignalHandlers();
+          runner.maybeCallDevToolsUriServiceExtension();
```

Comment:
There are several tests in the integration shard that connect to test projects. You should add one that verifies the URL is sent by running an app and then requesting the devtools url somehow

Reply:
Attempted to add tests. These do not pass locally. This future never completes, and I am not sure why: https://github.com/flutter/flutter/blob/705811fee2172b0f4bae1112a20901c731acb39c/packages/flutter_tools/lib/src/vmservice.dart#L712